File: tests/corpus/csharp/clean.cs
// Copyright (c) 2026 Example Corp. Licensed under the MIT license.

using System;

/// <summary>
/// Retries transient registry calls once with a short backoff.
/// </summary>
/// <param name="fetch">transport callback, invoked at most twice</param>
/// <returns>the raw record; null when the registry has no entry</returns>
class RetryClient {
    static string? Fetch(Func<string?> fetch) {
        try {
            return fetch();
        } catch (TimeoutException) {
            // workaround: the registry drops the first request under load
            return fetch();
        }
    }

    // ReSharper disable once InconsistentNaming
    const int BACKOFF_MS = 50; // tuned against the staging registry
}

File: tests/corpus/swift/clean.swift
// Copyright (c) 2026 Example Corp. Licensed under the MIT license.

import Foundation

/// Retries a transient registry call once with a short backoff.
///
/// - Parameter fetch: transport callback, invoked at most twice
/// - Returns: the raw record, or nil when the registry has no entry
/// - Throws: rethrows only non-transient transport errors
func fetchWithRetry(_ fetch: () throws -> String?) rethrows -> String? {
    do {
        return try fetch()
    } catch is TimeoutError {
        // workaround: the registry drops the first request under load
        return try fetch()
    }
}

// MARK: - Session state

// swiftlint:disable identifier_name
let kBackoffMs = 50 // tuned against the staging registry
// swiftlint:enable identifier_name

struct TimeoutError: Error {}

/// Errors produced during server trust evaluation.
enum TrustError2: Error {
    /// Certificate pinning failed.
    case certificatePinningFailed
    /// Public key pinning failed.
    case publicKeyPinningFailed
}
